lgc/patch/PatchReadFirstLane.cpp

```diff
@@ -397,16 +396,17 @@ void ReadFirstLaneOptimizer::findBestInsertLocation(const SmallVectorImpl<Instru
 
     for (;;) {
       const auto &mapIt = m_uniformDivergentUsesMap.find(current);
-      if (mapIt == m_uniformDivergentUsesMap.end())
-        break; // no further propagation possible
-
+      assert(mapIt != m_uniformDivergentUsesMap.end());
       const auto &divergentOperands = mapIt->second;
       if (divergentOperands.empty())
         break; // no further propagation possible
 
       if (divergentOperands.size() == 1) {
         // There is only a single operand, we can jump to it directly.
         current = divergentOperands[0];
+
+        if (!m_uniformDivergentUsesMap.count(current))
```

Comment:
I think you need to put this after the if/else to cover both cases.

Reply:
The else case has an inner loop with the invariant that `queue.size() >= 2` at the top of the loop; and queue members are only popped if they pass the m_uniformDivergentUsesMap check. Okay, but on second thought, `queue[0]` may end up going unchecked. So I think you're right.